examples/ruby/grpc-demo.gemspec

```diff
@@ -19,5 +19,6 @@ Gem::Specification.new do |s|
 
   s.add_dependency 'grpc', '~> 1.0'
 
-  s.add_development_dependency 'bundler', '~> 1.7'
+  s.add_development_dependency 'bundler',    '~> 1.7'
+  s.add_development_dependency 'googleauth', '>= 0.5.1', '< 0.7'
```

Comment:
why is `googleauth` being added as a development dependency here?

Reply:
Becuase googleauth is used by pubsub demo project I moved in examples/ruby directory.
Is it better to add a  production dependency by using`add_dependency`?

Reply:
The moving of the pubsub files is perhaps nice to have but isn't a necessary part of removing the `googleauth` production dependency (unless I'm missing something). Can we please revert those changes from this PR and keep this PR focused on dropping the `googleauth` dependency. Otherwise, I'm worried that, e.g., the change to the examples gemspec here might confuse people by appearing to be a new burden that the dropping of googleauth places on users.

Reply:
Feel free to move this files in a follow-up but I'd like to try keep this PR minimal

Reply:
I believe in it's a necessary change because pubsub files use googleauth gem at [here](https://github.com/grpc/grpc/blob/master/src/ruby/bin/apis/pubsub_demo.rb#L35).
pubsub files depend on googleauth and are included in grpc-ruby since grpc.gemspec specify at [here](https://github.com/grpc/grpc/blob/master/grpc.gemspec#L20).
If I leave it as it is, a situaction which  grpc-ruby doesn't depend on googleauth gem although it includes pubsub files depending on googleauth gem would cause.

Reply:
I see, you're right this feels like it belongs in a samples directory. I want to circle back to this after checking offline about the origin of these files (I'm actually not sure what/if they're used for now).

Also, is it possible to move these samples into its own subdirectory if `examples/ruby`?

thanks

Reply:
Do you mean that pubsub files should be positioned like under `examples/ruby/pubsub`? If so, it already is. Sorray, I'm not sure to understand it. Would you tell me more detail?

Reply:
My main goal is wondering if we can use a different gemspec file, so as to avoid adding a `googoelauth` dependency for the existing "route guide" and "greeter" examples, which don't need it.

Reply:
OK, I see. I put Gemfile for pubsub file not to let  "route guide" and "greeter" examples depned on googleauth.